js/undo-redo.js

```diff
@@ -2,62 +2,100 @@
 
 const MAX_HIST = 20;
 
+class Snapshot {
+  constructor(layerId, image, last) {
+    this.layerId = layerId;
+    this.width = curCanvasWidth;
+    this.height = curCanvasHeight;
+    if (last) {
```

Comment:
```suggestion

    if (last) {
```
А вот в этом файле наоборот мало пустых строк, объявление каких-либо начальных значений всегда лучше отделить пустой строкой перед непосредственными вычислениями